src/hardware/eeprom.rs

```diff
@@ -17,15 +24,15 @@ where
     // accomodate this, we repeat the I2C read for a set number of attempts with a fixed delay
     // between them. Then, we wait for the bus to stabilize by waiting until the MAC address
     // read-out is identical for two consecutive reads.
-    for _ in 0..40 {
+    for _ in 0..trys {
```

Comment:
I'd stick with 40 for everything here (and not pass `trys`, but keep the return type change).
Or is there a point in reducing it to 5 for driver?

Reply:
It tries for a whopping 4 seconds if Driver is not present then.. It should really work on the first try for Driver.

Reply:
40 i2c transaction attempts of one byte don't take 4 s at 100 kHz. Something else is wrong.

Reply:
Ah. I didn't see the 100 ms wait. Remove that then.

And if it should work the first time in your case, just use the respective function content. It's a single line and you don't need to use the repeated attempt logic.

Reply:
It says in the comments that it can take seconds on Stabilizer for the eeprom power to come good. 
I think I'll just leave this file unchanged then and inline one eeprom read.

Reply:
> It says in the comments that it can take seconds on Stabilizer for the eeprom power to come good.

But what's the point in waiting instead of continuing to try reading? It can't be synchronization and it can't be efficient use of resources or speed.

> I think I'll just leave this file unchanged then and inline one eeprom read.

Exactly. Less code.